### ISSUE TITLE
Multiple filters in a single layer

### ISSUE THREAD

**Issue by rudi77** [pyg-team/pytorch_geometric#1965]
## ❓ Questions & Help

Hi, I read the following paper ["An Invoice Reading System Using a Graph Convolutional Network"](https://link.springer.com/chapter/10.1007/978-3-030-21074-8_12). In this paper a GCN is used to extract certain information , like InvoiceDate etc. from invoices. The network consists of 5 layers. Each layer uses n chebychev filters of order k=3. The last layer predicts for each node of a document graph a certain class. I was wondering how I could implement such an architecture with pytoch-geometric. Is that possible at all? Looking at the geometric's nn.ChebConv documentation I think I would have to instantiate n ChebConv operators (I assume that one ChebConv operator represents one filter) in each hidden layer and then concatenate the results of each filter to a single (n * dim_in * dim_out) tensor which is then feed into the next layer and so on. Would this be a possible and valid solution or am I totally wrong. I would really appreciate if you could give me some tips/hints how I could implement such an architecture with your framework.

**Comment by rusty1s** [comment]
Sorry for late reply! I cannot access the PDF, but your description sounds totally fine to me. As far as I understand, in total, you have 5 * n ChebConv layers of order k=3.

**Comment by rudi77** [comment]
Hi, thank you for your response. They describe their GCN architecture as follows:
_"Proposed 5 layer GCN architecture for graph node classification. Input document graph is passed through 4 hidden layers, each followed by ReLU for non-linearity. We start initially with 16 Chebyshev filters (nf=16) increasing by a factor of 2 per layer..._" 
So the 'n' is different for each layer. I haven't mentioned that before. I think the intention behind this architecture is to have a model which generalizes well enough so that it is able to predict nodes with high accuracy in unseen graphs. For me as a newbie in this field, this makes totally sense. 
So I was asking myself why it is currently not possible to set the number of filters directly in the ChebyConv class. Is there a reason for that? 

Btw, I think this paper is not freely available but actually it is based on [Convolutional Neural Networks on Graphs with Fast Localized Spectral Filtering](https://arxiv.org/abs/1606.09375).

**Comment by rusty1s** [comment]
As far as I understand, the number of filters mentioned above corresponds to the number of `output_channels`. So, the above network architecture can be implemented via:
```python
conv1 = ChebConv(in_channels, 16, K=3)
conv2 = ChebConv(16, 32, K=3)
conv3 = ChebConv(32, 64, K=3)
conv4 = ChebConv(64, 128, K=3)
...
```

**Comment by rudi77** [comment]
Thanks for your prompt reply and for the example. I already suspected that I had misunderstood some, I assume, basic concepts.
